### PR TITLE
Support agent training, etc.

### PR DESCRIPTION
**Pull request by tastelikefeet**
1. Support sft_beta arg to DPOTrainer, which may stabilize the model generation.
2. Support Adalora and IA3
3. Support Agent training
4. Support dataset mixture
5. Support two new datasets: ms-agent and ms-bench
6. Support AnimateDiff merge lora
7. Fix some bugs in web-ui

### CODE REVIEW

File: swift/llm/agent/models/xverse.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        output_attentions=output_attentions,
+        output_hidden_states=output_hidden_states,
+        return_dict=return_dict,
+    )
+
+    hidden_states = outputs[0]
+    logits = self.lm_head(hidden_states)
+
+    loss = None
+    if labels is not None:
```
Comment (Jintao-Huang) [inline swift/llm/agent/models/xverse.py]: loss有可能在compute_loss中计算不, 这样所有的模型都可以支持了

File: swift/llm/utils/argument.py [hunk 2]
```diff
@@ -90,10 +97,32 @@ class SftArguments:
     lora_alpha: int = 32
     lora_dropout_p: float = 0.05
     lora_bias_trainable: Literal['none', 'all'] = 'none'
+    use_rslora: bool = False
```
Comment (Jintao-Huang) [inline swift/llm/utils/argument.py:100]: 参数有点多, 要不全部重构一下, 这块

File: swift/llm/utils/argument.py [hunk 3]
```diff
@@ -173,6 +202,8 @@ class SftArguments:
     only_save_model: Optional[bool] = None
     neftune_alpha: Optional[float] = None
 
+    gpu_memory_fraction: Optional[float] = None
```
Comment (Jintao-Huang) [inline swift/llm/utils/argument.py:203]: 文档补充一下吧

File: swift/llm/utils/preprocess.py [hunk 4]
```diff
@@ -118,7 +118,7 @@ def __call__(self, dataset: HfDataset) -> HfDataset:
                 response.append(conversations[-1][self.value_key])
                 system.append(sys)
                 history.append(h)
-            except AssertionError:
+            except Exception:
```
Comment (Jintao-Huang) [inline swift/llm/utils/preprocess.py]: 这里是为什么呀

File: swift/llm/utils/template.py [hunk 5]
```diff
@@ -7,6 +7,8 @@
 from torch import Tensor
 from transformers import PreTrainedTokenizerBase, StoppingCriteria
 
+from swift.llm.agent.utils import calculate_loss_scale
```
Comment (Jintao-Huang) [inline swift/llm/utils/template.py:13]: 这里看不清楚 要不merge main一下

**Comment by yingdachen** [comment]
plus also update the pr title and commit msg

**Review by Jintao-Huang — APPROVED** [review]

File: swift/llm/agent/models/xverse.py (hunk 1 above)
Comment (tastelikefeet) [reply swift/llm/agent/models/xverse.py]: done

File: swift/llm/utils/argument.py (hunk 2 above)
Comment (tastelikefeet) [reply swift/llm/utils/argument.py:100]: 后面一起重构

File: swift/llm/utils/argument.py (hunk 3 above)
Comment (tastelikefeet) [reply swift/llm/utils/argument.py:203]: 已补充文档

File: swift/llm/utils/preprocess.py (hunk 4 above)
Comment (tastelikefeet) [reply swift/llm/utils/preprocess.py]: 已回滚

**Pull request merged**